features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/SessionEndedMetricDispatcher.kt

```diff
@@ -51,12 +53,32 @@ internal class SessionEndedMetricDispatcher(private val internalLogger: Internal
         }
     }
 
-    override fun onSdkErrorTracked(sessionId: String, errorKind: String?) {
```

Comment:
cannot we have a situation when we stopped session A and started session B, but due to async nature of different pipelines in the SDK, we have this method called with error coming from session A by the time `startMetric` was called with Session B? not sure what is the probability of that though.

Reply:
I am not sure the probability of this neither, but with this pull request, the error coming from Session A will belongs to Session B since `startMetric` has called. I am not sure if we have anyway to improve this, since `endMetric` of Session A has been called, the telemetry is sent out, the error will not be able to catch the train which has left.

Reply:
In theory we can have a skew in the metric data we receive in such case, so maybe it is better to rollback this change and still pass `sessionId` to this method from the `DatadogContext` we have in the `TelemetryEventHandler`?

Reply:
I have two questions over this:
1. even if we use session Id to track the error, the error count in session B can be correct, but in the session A will always be less than the correct one since the metric can be sent out, is that something we can tolerate?
2. I tried to use `sdkCore.getFeature(Feature.RUM_FEATURE_NAME)?.withWriteContext` that you suggested to retrieve the session id for the error, I got always NULL_UUID, is that normal?

Reply:
1. Don't know honestly, depends on the further metric usage and if we can have falsy data because of that.
2. You don't need to use it directly, you can hook inside the lambda in the `TelemetryEventHandler` or you can use  something like `RumContext.fromFeatureContext(sdkCore.getDatadogContext().featuresContext[Feature.RUM_FEATURE_NAME])`. `NULL_UUID` means that session is not yet created.